### ISSUE TITLE
react importeted components and esbuild or swc

### ISSUE THREAD

**Issue by KoderFPV** [theKashey/react-imported-component#229]
Hi, 
Is there a possibility to use this library with other bundlers or other loaders to speed up the process of building and rebuilding applications?

In my understanding, this library is dependent on babel because we need babel plugins to get it to work, right?

**Comment by theKashey** [comment]
Correct. It can "work" with esbuild/swc by calling babel only for the files with dynamic imports, which is just a few files among others, maintaining build speed and not losing functionality.

The corresponding plugin can be done in the form of webpack or rollup(vite) loader - checking for a keywords/import and running targeted babel transform if needed.

**Comment by KoderFPV** [comment]
@theKashey in our architecture we have about 250 (sic!) lazy loaded components. We put all lazy imports in to few files. Lets say `commonLazyImports` and `themedLazyImported` (we have support for few themes in our app).

Do we have to transfer only files with usage of lazy import, or also in the places where we use components.

Ex structure

### commonLazyImport.tsx
```
const component = lazy(() => import('./component')
```

### OtherComponents.tsx

```
import Component from `./commonLazyImport.tsx`

export const OtherComponent = () => <Component />
```

So do we have to transform both `commonLazyImport.tsx` and `OtherComponents.tsx`?

**Comment by theKashey** [comment]
Only files with dynamic imports need the babel pass, that is `commonLazyImport` and index file (the one generated by `imported-components` command)

Also - babel is required only for SSR, for clientside-only activities it is not required at all.

**Comment by KoderFPV** [comment]
Ok finaly we did, however right after we realized that our build time does not have problems with ts(x) files :)

However I will live it for someone who also will decided to use `react-imported-components` with `esbuild-loader` or `swc-loader` 

### About app
Our apliaction has theming mechanism. Its not only about CSS, but also about HTML structure and logic. 
So we decide to load tsx files related to other themes by lazy loading from this library.

### File structure
So in our architecture we have couple files related to `react-imported-component` library. 
- lazyImports.tsx <- we have here all lazy imports common for whole app
- firstThemeLazyImports.tsx <- we have here all lazy imports for first theme
- secondThemeLazyImports <- we have second theme lazy imports

Also we are generating by `imported-components` command extra files:
- common-chunks.ts <- for common lazy imports
- firstTheme-chunks.ts <- for first theme lazy imports
- secondTheme-chunks.ts <- for second theme imports

### Example webpack laoders config **client**

`
rules = [
    {
        test: /\.[jt]sx?$/,
        exclude: /node_modules/,
        loader: 'esbuild-loader',
    },
    {
        test: /(lazyImports|firstThemeLazyImports|secondThemeLazyImports|common-chunks|firstTheme-chunks|secondTheme-chunks)\.tsx?$/,
        loader: 'babel-loader',
        options: {
            plugins: ['react-imported-component/babel'],
        },
    },
    // this lib has problem with esbuild so should be done by babel
    {
        test: /node_modules\/yaml\/browser\/dist\/.*/,
        type: 'javascript/auto',
        use: {
            loader: 'babel-loader',
            options: {
                presets: ['@babel/preset-env'],
            },
        },
    },
];
`

**Comment by theKashey** [comment]
Problem with babel can be resolved via customer loader like
```tsx
// chatGpt generated
const babelLoader = require('babel-loader');

module.exports = function(source, sourceMap) {
  // Check if the source code contains the magic string
  const hasMagicString = source.includes('react-imported-component'); // ⬅️⬅️

  if (hasMagicString) {   
    // pass down to babel
    return babelLoader.call(this, source, sourceMap);
  }

  // If the magic string is not present, return the original source code
  return [source, inputSourceMap]
};
```

In webpack config you can apply this loader to all js/ts files passing babel configuration into options and it will execute babel transform only for the files that needed it.

The only "cost" you have to pay - source.includes - which is "nothing"
